api/src/main/java/org/commonjava/indy/util/LocationUtils.java

```diff
@@ -55,6 +59,7 @@ public static KeyedLocation toLocation( final ArtifactStore store )
         KeyedLocation location = (KeyedLocation) store.getTransientMetadata( KEYED_LOCATION_METADATA );
         if ( location != null )
         {
+            location.setAttribute( Location.AS_DIRECTORY,asDir );
```

Comment:
Okay, I just caught this. We're caching the Location instances in the ArtifactStore (maybe that's not a great idea, I'm not sure). That means those Location instances get used for all types of access to content.

If we access content one time with `asDir == true`, then will the caching of this Location object **force all content access after this to use the `asDir == true` attribute too?** It seems like we may be switching off our ability to route directory requests to `package.json` here.

Keep in mind that the Location caching is global to the Indy instance, which means it will affect concurrent access to the content or access at any point in the future for that ArtifactStore, as long as the transient metadata exists (it's an in-memory construct).

Reply:
This makes sense, so this AS_DIRECTORY or RAW_VIEW attr is not a Location or ArtifactStore level concept but a path or ConcreteResource level concept. It's not good idea to use it here. I'll close this pr and try other solutions.